PowerEditor/src/WinControls/WindowsDlg/WindowsDlg.cpp

```diff
@@ -335,34 +353,20 @@ INT_PTR CALLBACK WindowsDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM lPa
 					if (pLvdi->item.mask & LVIF_TEXT)
 					{
 						pLvdi->item.pszText[0] = 0;
-						size_t index = pLvdi->item.iItem;
-						if (index >= _pTab->nbItem() || index >= _idxMap.size())
+						Buffer* buf = getBuffer(pLvdi->item.iItem);
+						if (!buf)
 							return FALSE;
-						index = _idxMap[index];
-
-						//const Buffer& buffer = _pView->getBufferAt(index);
-						BufferID bufID = _pTab->getBufferByIndex(index);
-						Buffer * buf = MainFileManager.getBufferByID(bufID);
+						generic_string text;
 						if (pLvdi->item.iSubItem == 0) // file name
 						{
-							int len = pLvdi->item.cchTextMax;
-							const TCHAR *fileName = buf->getFileName();
-							generic_strncpy(pLvdi->item.pszText, fileName, len-1);

```

Comment:
There was a buffer overrun that was revealed by my changes.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/strncpy-strncpy-l-wcsncpy-wcsncpy-l-mbsncpy-mbsncpy-l?view=msvc-160
It seems, in some cases more characters were copying than the length of the `pszText` (here or below). 
This led to some other memory being overwritten which resulted in a crash in Scintilla code:
![image](https://user-images.githubusercontent.com/9664141/126214396-849e7db3-e520-4ac6-ae03-a87cbdf91424.png)

Now the code uses generic_string that allocates memory dynamically and is safer than string copy functions.
The only time string copy is called is at line 401 with the proper size check.

Reply:
It was done in a separate commit to make it easier for review. So that it can be reviewed separately.